pinot-common/src/test/java/org/apache/pinot/common/utils/SegmentUtilsTest.java

```diff
@@ -35,4 +47,32 @@ public void testGetSegmentCreationTimeMs() {
     segmentZKMetadata.setPushTime(2000L);
     assertEquals(SegmentUtils.getSegmentCreationTimeMs(segmentZKMetadata), 2000L);
   }
+
+  @Test
+  public void testGetUploadedRealtimeSegmentPartitionId() {
```

Comment:
Good to add a UT that checks for `SegmentPartitionMetadata.getUploadedSegmentPartitionId` -1 when it is not externally partitioned.